pagopa/io-app
# fix(IDPay): [IODPAY-119,IODPAY-120] Fix instruments enrollment UI not following design

## Short description
update `InstrumentsEnrollmentScreen.tsx` to more strictly follow the design

<img src="https://user-images.githubusercontent.com/60693085/218091602-84a3fe8b-6154-454f-bd13-218b369706df.png" width=200/>

## List of changes proposed in this pull request
- added card icon and eventual fallback 
- added typography symbols for the maskedPan
- added wallet icon in footer
- fixed typo in i18n
- exchanged some legacy nativeBase components for their react-native counterparts
- updated footer text  typography component to follow design


## How to test
Having at least one enrolled initiative, navigate to its detail page, then to the "add/modify payment methods" screen, and verify that it renders correctly, which in this case means that it has an icon or image for each card and its footer correctly displays the specified text and the wallet icon.
